source/Debugger/Debug.cpp

```diff
@@ -6538,7 +6538,9 @@ bool ParseAssemblyListing ( bool bBytesToMemory, bool bAddSymbols )
 		{
 			*p = 0;
 			//	sscanf( sLine, "%s %s %s %s %s %s %s %s", sAddr1, sByte1, sByte2, sByte3, sLineN, sLabel, sAsm, sParam );
-			sscanf( sLine, "%X", &nAddress );
+			unsigned nScanAddress; // helper variable in case sizeof(DWORD) != sizeof(unsigned)
+			sscanf( sLine, "%X", &nScanAddress );
```

Comment:
I suspect this will require more care.

I must define `DWORD` as `unsigned int`. Can you check what your version of gcc uses?

We will probably have to use a macro to select the correct print format like we already do for `size_t` and `ptrdiff`

Reply:
Based on my comments [here](https://github.com/AppleWin/AppleWin/pull/1346#discussion_r1865124798), mingw uses `unsigned long int` for `DWORD`.

By `sscanf`ing into a secondary variable, we know exactly which format to use (`"%X"` is appropriate for `unsigned` - see [here](https://en.cppreference.com/w/c/io/fscanf)). The assignment then converts that to whatever `DWORD` is defined as.